acceptance/suites/tests/555-puppetserver-service/reboot.rb

```diff
@@ -0,0 +1,55 @@
+if (options[:type] == 'pe') then
+  pssn='pe-puppetserver'
+else
+  pssn='puppetserver'
+end
+
+test_name "Validates the ability of the #{pssn} service to reliably start after a reboot."
+
+max=3
+sleeptime=45
+retryCounter=0
+step "\n-----------\nValidate #{pssn} is up and running, then reboot, repeat #{max} times\n-----------"
+
+for i in 1..max 
+  step "\n-----\nLoop #{i}\n-----"
+   
+  step "check uptime"
+  origUptime=on(master, "awk '{print $1}' /proc/uptime").stdout.chomp.to_i
+
+  step "reboot"
+  #rebooting with on(master,'reboot') causes an IOError: stream end.  This seems like a sane workaround.
+  #QENG-1914 for more...
+  system "ssh -i #{options[:keyfile]} -o StrictHostKeyChecking=no root@#{master.reachable_name} \'reboot\'"
```

Comment:
> Beyond reasons of good standard practice and concerns about the ssh command working in MacOS and in CI, is there a compelling reason why we should continue to struggle with the typical beaker method in this case that I'm missing?

Yes, my primary concern is trying to make this robust for the long-term.  For example, even though it works today in all of the places we typically run beaker, we're introducing a lot of variation into the system that needs to be maintained over a long time horizon.  The variation I see is this.

Today, we have one transport method, Net::SSH and specifically the beaker methods that wrap around Net::SSH.  If, in the future, we find a reason to change the behavior of this transport then we have only one place to fix it and then validate the change across all of the places we run beaker.

If we introduce another transport mechanism that calls directly down to the system and there's a change we need to make in the future, we now have many more things we need to make behave.  We'll have to update beaker's wrappers around Net::SSH and in addition make sure these direct calls to the system ssh command work on all systems we run beaker from.

It's not so much about the risk of this breaking in the future, it's about the increase in effort it will take to change the behavior in the future.  To this end, the fact that it works today doesn't really address my concern, because the concern is more, "how much effort will it take to maintain this behavior over the long haul?"

> Is there a way we can configure Ruby's Net::SSH so that it will gracefully accept that SSHD is shutting down?

I think there's a way to gracefully handle that SSHD terminates, but I don't think it involves reconfiguring Net::SSH.

I'm going to try and provide an example of catching the exception that's thrown, which is probably the most direct and un-surprising way to handle this sort of situation.

I'll follow up here with what I find out.